Dpad mapped to toggle shooter and intake

andrewmartin5 opened the pull request:
Dpad Up: Shooter on slow speed
Dpad Down: Shooter off
Dpad Left: Deploy Intake
Dpad RIght: Retract Intake

andrewmartin5 committed:
Dpad now controls the shooter

shooter spins at decreased speed instead of stopping

andrewmartin5 committed:
Dpad Mapped to intake

left = deploy; right = retract

andrewmartin5 committed:
Dpad Trigger now includes support for noncardinal directions

msoucy commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -278,20 +279,27 @@ public CommandBase visionAlignment() {
         driveController.getButton(Button.kBumperRight).whenHeld(drive.slowTurn(true));
         driveController.getButton(Button.kBumperLeft).whenHeld(drive.slowTurn(false));
-
+        final DpadTrigger stopShooter = DpadTrigger.DpadDown(driveController);
```
Why make the triggers just to use them once in the next line? Combine them please

msoucy commented on src/main/java/frc/robot/triggers/DpadTrigger.java:
```diff
@@ -0,0 +1,102 @@
+     * @param controller The controller to use
+     * @param angle      The angle of dpad button (0 is up)
+     */
+    public DpadTrigger(XboxController controller, int angle) {
```
Can all these angles just be a nice enum or similar, instead of having tons of nearly identical functions?

msoucy commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -128,23 +128,29 @@ public boolean isFinished() {
     }
 
-    public CommandBase spinDown() {
+    public CommandBase SlowSpin() {
```
Why is that capitalized?

andrewmartin5 committed:
DpadTrigger Simplified with enum

msoucy committed:
Updated dependency so POVButton can be used

msoucy commented:
This needs a rebase

bot190 committed:
Use ButtonXboxController POV button implementation

Replaces the DpadTrigger with the ButtonXboxController implementation that uses the POVButton class from WPILib.
